### PR TITLE
Jest testing/setup

### PR DESCRIPTION
**Pull request by ZeeshaanMaudar**
### Scope
Set up testing environment to use Jest and Enzyme

### Work Done
 - Installed all required files as per Gatsby docs (might not need mocks but put it there just in case)
 - Check testing through a simple snapshot test of the Header component.
 - Test successful

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,13 +25,28 @@
...
+    "babel-core": "^7.0.0-bridge.0",
+    "babel-jest": "^23.6.0",
+    "babel-preset-gatsby": "^0.1.5",
+    "enzyme": "^3.7.0",
+    "enzyme-adapter-react-15": "^1.2.0",
+    "enzyme-adapter-react-16": "^1.7.0",
+    "identity-obj-proxy": "^3.0.0",
+    "jest": "^23.6.0",
+    "prettier": "^1.14.2",
+    "react-test-renderer": "^16.6.3",
```
Comment (schalkventer) [inline package.json:40]: Are you using Prettier? If not, then we can remove it.

File: setupTests.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+var enzyme = require('enzyme');
```
Comment (schalkventer) [inline setupTests.js]: No reason to not have this as ES6. Rather use `const`. :)

File: package.json (hunk 1 above)
Comment (ZeeshaanMaudar) [reply package.json:40]: Not anymore. Eslint is set up now. I removed it and its file too.

File: setupTests.js (hunk 2 above)
Comment (ZeeshaanMaudar) [reply setupTests.js]: Done

**Pull request merged**